### PR TITLE
Fix build CTL from external project

### PR DESCRIPTION
**Pull request by ghost**
# What this PR does?
* Fixes cmake configure file and compilation warning

# How to test
* `mkdir build`
* `cd build`
* `cmake  ..`
* `make`
* cd `bin` and run all test executables

### CODE REVIEW

File: lib/IlmCtl/CtlRcPtr.cpp [hunk 1]
```diff
@@ -96,7 +96,7 @@ throwRcPtrExc (const RcObject *lhs, const RcObject *rhs)
 Mutex &
 rcPtrMutex (RcObject *ptr)
 {
-    return mutexes[reinterpret_cast<unsigned long>(ptr) % NUM_MUTEXES];
+    return mutexes[reinterpret_cast<unsigned long long int>(ptr) % NUM_MUTEXES];
```
Comment (yuriyvolkov) [inline lib/IlmCtl/CtlRcPtr.cpp:99]: @AntonQubecinema I've checked this on 32-bit `Linux ubuntu 4.4.0-53-generic #74-Ubuntu SMP Fri Dec 2 15:58:04 UTC 2016 i686 i686 i686 GNU/Linux` and included tests seem to pass

**Review by yuriyvolkov — CHANGES_REQUESTED** [review]
when I use out of source builds say in the `build` I end up with the directory structure like this in it (note `deps` and `build/deps`)
```
.
├── bin
├── build
│   └── deps
│       ├── acescontainer
│       └── openexr
├── CMakeFiles
│   ├── 3.5.1
│   │   ├── CompilerIdC
│   │   └── CompilerIdCXX
│   └── CMakeTmp
├── deps
│   ├── acescontainer
│   │   └── config
│   └── openexr
│       ├── Contrib
│       ├── IlmBase
│       ├── OpenEXR
│       ├── OpenEXR_Viewers
│       └── PyIlmBase
├── doc
│   └── CMakeFiles
├── lib
│   ├── CMakeFiles
│   │   └── CTL.dir
│   ├── dpx
│   │   └── CMakeFiles
│   ├── IlmCtl
│   │   └── CMakeFiles
│   ├── IlmCtlMath
│   │   └── CMakeFiles
│   └── IlmCtlSimd
│       └── CMakeFiles
├── OpenEXR_CTL
│   └── CMakeFiles
└── unittest
    ├── CMakeFiles
    │   └── check.dir
    ├── IlmCtl
    │   └── CMakeFiles
    └── IlmCtlMath
        └── CMakeFiles


```

**Comment by ghost** [comment]
@yuriyvolkov build structure fixed

File: lib/IlmCtl/CtlRcPtr.cpp (hunk 1 above)
Comment (ghost) [reply lib/IlmCtl/CtlRcPtr.cpp:99]: OK

**Review by yuriyvolkov — APPROVED** [review]

File: lib/IlmCtl/CtlRcPtr.cpp (hunk 1 above)
Comment (sivachandran) [reply lib/IlmCtl/CtlRcPtr.cpp:99]: @yuriyvolkov Wondering how did you realise this issue?

**Review by RajaBoseLeoM — APPROVED** [review]
👍

**Review by sivachandran — APPROVED** [review]

**Pull request merged**